ProGetAutomation/Functions/Get-ProGetAssetContent.ps1

```diff
@@ -6,37 +6,47 @@ function Get-ProGetAssetContent
+    The `Get-ProGetAssetContent` function gets an asset's content from ProGet. Pass the name of the root asset directory
+    to the `DirectoryName` parameter. Pass the path to the asset to the `Path` parameter. If the URL to an asset
+    directory in ProGet is `https://proget.example.com/assets/versions/subdirectory/file`, the directory parameter is
+    the first directory after `assets/` (in this example `versions`). The `Path` parameter would be the rest of the url
+    in this case `subdirectory/file`.
 
     If an asset doesn't exist, an error will be written and nothing is returned.
 
-    Pass a ProGet session object to the `Session` parameter. This object controls what instance of ProGet to use and what credentials and/or API keys to use. Use the `New-ProGetSession` function to create session objects.
+    Pass a ProGet session object to the `Session` parameter. This object controls what instance of ProGet to use and
+    what credentials and/or API keys to use. Use the `New-ProGetSession` function to create session objects.
 
     .Example
     Get-ProGetAssetContent -Session $session -DirectoryName 'versions' -Path 'subdirectory/file.json'
 
-    Demonstrates how to get the contents of an asset. In this case, the `subdirectory/file.json` asset's contents in the `versions` asset directory is returned.
+    Demonstrates how to get the contents of an asset. In this case, the `subdirectory/file.json` asset's contents in the
+    `versions` asset directory is returned.
     #>
     param(
-        [Parameter(Mandatory = $true)]
-        [Object]
-        # A session object that represents the ProGet instance to use. Use the `New-ProGetSession` function to create session objects.
-        $Session,
+        # A session object that represents the ProGet instance to use. Use the `New-ProGetSession` function to create
+        # session objects.
+        [Parameter(Mandatory)]
+        [Object] $Session,
 
-        [Parameter(Mandatory = $true)]
-        [string]
         # The name of the asset's asset directory.
-        $DirectoryName,
+        [Parameter(Mandatory)]
+        [String] $DirectoryName,
 
-        [string]
         # The path to the file in the asset directory, without the asset directory's name.
-        $Path
+        [String] $Path
     )
 
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
     $uri = '/endpoints/{0}/content/{1}' -f $DirectoryName,$Path
 
-    return Invoke-ProGetRestMethod -Session $Session -Path $uri -Raw
+    $response = Invoke-ProGetRestMethod -Session $Session -Path $uri -Raw
+    if (-not $response)
+    {
+        return
+    }
+
+    return ,$response.Content

```

Comment:
I recently discovered that `Write-Output` has a `-NoEnumerate` switch. Does that behave the same? I.e. can we do:

```powershell
return $response.Content | Write-Output -NoEnumerate
```

Reply:
From the help: 

>    [!IMPORTANT] There is an issue with this switch in Windows PowerShell that is fixed in PowerShell 6.2 and 
above. When using NoEnumerate and explicitly using the InputObject parameter, the command > still enumerates. To work around this, pass the InputObject argument(s) positionally.

![image](https://github.com/webmd-health-services/ProGetAutomation/assets/1082153/bf085d1a-8a96-4ff9-a91f-d79b4a01fb0f)

Reply:
😞